src/storage/config.rs

```diff
@@ -110,10 +113,43 @@ impl Default for Config {
 }
 
 impl Config {
+    fn validate_engine_type(&mut self) -> Result<(), Box<dyn Error>> {
+        let v1_kv_db_path =
+            config::canonicalize_sub_path(&self.data_dir, DEFAULT_ROCKSDB_SUB_DIR).unwrap();
+        let v2_tablet_path =
+            config::canonicalize_sub_path(&self.data_dir, DEFAULT_TABLET_SUB_DIR).unwrap();
+
+        let kv_data_exists = Path::new(&v1_kv_db_path).exists();
+        let v2_tablet_exists = Path::new(&v2_tablet_path).exists();
+        if kv_data_exists && v2_tablet_exists {
+            return Err("The data dir has subfolders for both RaftKv and PartitionRaftKv".into());
+        }
+
+        // v1's data exists, but the engine type is v2
+        if kv_data_exists && self.engine == EngineType::RaftKv2 {
+            info!(
+                "TiKV has data for RaftKv engine but the engine type in config is Partition-Raft-Kv. Ignore the config and keep Raft-Kv instead"
```

Comment:
Ditto, use raft-kv and partitioned-raft-kv instead as they are the actual config names.